apps/src/lab2/types.ts

```diff
@@ -260,3 +260,11 @@ export interface ExtraLinksData {
   can_delete: boolean;
   level_name: string;
 }
+
+export const TEXT_BASED_LABS: AppName[] = [
+  'applab',
+  'javalab',
+  'weblab',
+  'pythonlab',
+  'weblab2',
```

Comment:
Is this an exhaustive list? Are applab and weblab compatible with Lab2? If "not yet", maybe we should also include gamelab? Not sure if there are other worth considering too, like ai lab.

Alternatively, I wonder if it might be more robust to actually check whether the level uses Blockly or not.

Reply:
No this was my loose list :) I put labs that I knew were text-based, and included those not on lab2 so it wasn't something we had to worry about in the future. But you are probably right that having a half-correct list isn't great. I can update this for now to just be lab2 labs that are text based, and when a lab is migrated when can add it to the list. I'm not sure if we can check consistently if a level uses Blockly, is that a setting on every lab type?

Reply:
All Blockly levels, including Music, should extend the Blockly level class. I've seen us do `level.is_a? Blockly` in some places. I'm not sure if that's helpful or not. 
I'd vote for either being robust like this (if it works) or having a short list that just has supported lab2 labs if that doesn't work.

Reply:
We can't do that on the frontend without sending that as a level property (which I don't think we do now). I'll pare down the list and comment that it's only lab2 labs.